Quantum_Circuit_Design/dqn.py

```diff
@@ -74,30 +73,30 @@ def _circuit_to_state(self, circuit: QuantumCircuit) -> np.ndarray:
 
             # Flatten the unitary matrix and normalize
             unitary_array = np.asarray(unitary).flatten()
-            print(len(unitary),len(unitary_array))
-            return unitary
+            return unitary_array
 
     def reset(self):
         self.circuit = QuantumCircuit(self.num_qubits)
         return self._circuit_to_state(self.circuit)
 
+    def generate_actions(self,num_qubits):
+        gates = [HGate(), SGate()]  # Single-qubit gates
```

Comment:
Is there no need for the rest of the previously hardcoded gates (TGate(), XXGate(), YGate(), ZGate())?